IdentityCore/src/requests/MSIDInteractiveTokenRequest.m

```diff
@@ -276,18 +276,36 @@ - (void)acquireTokenWithCode:(NSString *)authCode
             return;
         }
         
-        BOOL accountChecked = [self.tokenResponseValidator validateAccount:self.requestParameters.accountIdentifier
-                                                               tokenResult:tokenResult
-                                                             correlationID:self.requestParameters.correlationId
-                                                                     error:&validationError];
+        void (^validateAccountAndCompleteBlock)(void) = ^
+        {
+            NSError *validationError;
+            BOOL accountChecked = [self.tokenResponseValidator validateAccount:self.requestParameters.accountIdentifier
+                                                                   tokenResult:tokenResult
+                                                                 correlationID:self.requestParameters.correlationId
+                                                                         error:&validationError];
+            
+            if (!accountChecked)
+            {
+                completionBlock(nil, validationError, nil);
+                return;
+            }
+            
+            completionBlock(tokenResult, nil, nil);
+        };
         
-        if (!accountChecked)
+#if TARGET_OS_OSX
```

Comment:
should we only have this logic in default requests?
This class will be also used in ADAL on macOS that doesn't need this logic.

Reply:
done